miio/curtain_youpin.py

```diff
@@ -0,0 +1,270 @@
+
+    @property
+    def target_position(self) -> int:
+        """Target position
+        Value range: [0, 100, 1]
+        """
+        return self.data["target_position"]
+
+    @property
+    def adjust_value(self) -> int:
+        """ Adjust value
+        Value range: [-100, 100, 1]
+        """
+        return self.data["adjust_value"]
+
+    def __repr__(self) -> str:
+        s = (
+            "<CurtainStatus"
+            "status=%s,"
+            "polarity=%s,"
+            "position_limit=%s,"
+            "night_tip_light=%s,"
+            "run_time=%s,"
+            "current_position=%s,"
+            "target_position=%s,"
+            "adjust_value=%s>"
+            % (
+                self.status,
+                self.polarity,
+                self.position_limit,
+                self.night_tip_light,
+                self.run_time,
+                self.current_position,
+                self.target_position,
+                self.adjust_value
+            )
+        )
+        return s
+
+class CurtainMiot(MiotDevice):
+    """Main class representing the lumi.curtain.hagl05 curtain."""
+
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+    ) -> None:
+        super().__init__(_MAPPING, ip, token, start_id, debug, lazy_discover)
+
+    @command(
+        default_output=format_output(
+            "",
+            "Device status: {result.status}\n"
+            "Manual enabled: {result.manual_enabled}\n"
+            "Polarity: {result.polarity}\n"
+            "Position limit: {result.position_limit}\n"
+            "Enabled night tip light: {result.night_tip_light}\n"
+            "Run time: {result.run_time}\n"
+            "Current position: {result.current_position}\n"
+            "Target position: {result.target_position}\n"
+            "Adjust value: {result.adjust_value}\n",
+        )
+    )
+    def status(self) -> CurtainStatus:
+        """Retrieve properties."""
+
+        return CurtainStatus(
+            {
+                prop["did"]: prop["value"] if prop["code"] == 0 else None
+                for prop in self.get_properties_for_mapping()
+            }
+        )
+
+    @command(
+        click.argument("motor_control", type=EnumType(MotorControl)),
+        default_output=format_output("Set motor control to {motor_control}"),
+    )
+    def set_motor_control(self, motor_control: MotorControl):
+        """Set motor control.
+        """
+         self.set_property("motor_control", motor_control.value)
+
+    @command(
+        click.argument("target_position", type=int),
+        default_output=format_output("Set target position to {target_position}"),
+    )
+    def set_target_position(self, target_position: int):
+        """Set target position.
+        """
+        if target_position < 0 or target_position  > 100:
+            raise ValueError("Wrong value")
```

Comment:
```suggestion
            raise ValueError("Value must be between ]0,100[ value, was %s" % target_position)
```
Assuming 0 and 100 are not valid values, otherwise the code needs to be adapted and [0,100] is the range.